flatdata-rs/lib/src/structs.rs

```diff
@@ -31,76 +31,39 @@ use std::fmt::Debug;
 #[doc(hidden)]
 pub use std::marker;
 
-/// A type in flatdata used for reading data.
-///
-/// Each struct reference in generated code implements this trait.
-pub trait Ref: Clone + Copy + Debug + PartialEq {}
-
-/// A mutable type in flatdata used for writing data.
-///
-/// Each struct reference in generated code has a corresponding type with suffix
-/// `Mut` which implements this trait.
-pub trait RefMut: Debug {}
-
 /// A factory trait used to bind lifetime to Ref implementations.
 ///
 /// Vector/ArrayView-like classes cannot be directly implemented over the
 /// structs since that binds lifetime too early. Instead this generic factory
 /// and Higher-Rank-Trait-Bounds are used to emulate higher-kinded-generics.
-pub trait Struct<'a>: Clone {
+pub trait Struct: 'static + Debug + Sized {
+    /// Create a new struct
```

Comment:
1. First line should be a paragraph (i.e. empty line after is needed).
2. There is a standard `# Safety` section, https://doc.rust-lang.org/book/ch14-02-publishing-to-crates-io.html?highlight=documentation#commonly-used-sections.

Reply:
Done